api/src/main/java/org/apache/gravitino/tag/SupportsTags.java

```diff
@@ -58,6 +59,8 @@ public interface SupportsTags {
    * @param tagsToAdd The arrays of tag name to be added to the object.
    * @param tagsToRemove The array of tag name to be removed from the object.
    * @return The array of tag names that are associated with the object.
+   * @throws TagAlreadyAssociatedException If the tag is already associated with the object.
    */
-  String[] associateTags(String[] tagsToAdd, String[] tagsToRemove);
+  String[] associateTags(String[] tagsToAdd, String[] tagsToRemove)
```

Comment:
So, what's the behavior if the tag to be removed does not exist?

Reply:
nothing happened.

Reply:
Add more comments about this.